nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -211,6 +200,50 @@ public static IEnumerable<Dependency> GetTopLevelPackageDependenyInfos(Immutable
         }
     }
 
+    /// <summary>
+    /// Given an MSBuild string and a set of properties, returns our best guess at the final value MSBuild will evaluate to.
+    /// </summary>
+    /// <param name="msbuildString"></param>
+    /// <param name="propertyInfo"></param>
+    /// <returns></returns>
+    public static string GetRootValue(string msbuildString, Dictionary<string, string> propertyInfo)
+    {
+        bool modified;
+        do
+        {
+            modified = false;
+            if (HasPropertyReplace(msbuildString))
+            {
+                modified = false;
+                foreach (var property in propertyInfo)
+                {
+                    var propertyReplacement = $"$({property.Key})";
+                    if (msbuildString.Contains(propertyReplacement))
+                    {
+                        msbuildString = msbuildString.Replace(propertyReplacement, property.Value);
+                        modified = true;
+                        break;
+                    }
+                }
+            }
+        } while (modified is true);
+
+        return msbuildString;
+    }
+
+    public static bool HasPropertyReplace(string versionContent)
+    {
+        return versionContent.Contains("$(") && versionContent.Contains(')');
```

Comment:
As a sanity check you could get the index of both and ensure that they are ordered properly.